sound/soc/sof/pm.c

```diff
@@ -82,6 +85,11 @@ static int sof_restore_pipelines(struct snd_sof_dev *sdev)
 						 comp_dai->comp.hdr.cmd,
 						 comp_dai, sizeof(*comp_dai),
 						 &r, sizeof(r));
+
+			config = dai->dai_config;
+			if (config->dai_index > max_index)
+				max_index = config->dai_index;
+
```

Comment:
should you check that there is at least one DAI in this pipeline as well? if nothing happens you will still allocate a link array of 1 element.

Reply:
yes, I will add it.